app/policies/synapse_policy.rb

```diff
@@ -38,7 +38,7 @@ def update?
   end
 
   def destroy?
-    record.user == user || admin_override
+    record.mappings.empty? && (record.user == user || admin_override)
```

Comment:
maybe we should add some custom text as a disclaimer about this in the API documentation?